lib/active_record_inherit_assoc.rb

```diff
@@ -14,6 +14,18 @@
   # I went with the first one out of simplicity.
   ActiveRecord::Associations::Builder::Association::VALID_OPTIONS << :inherit
   ActiveRecord::Associations::Builder::Association::VALID_OPTIONS << :inherit_allowed_list
+when 6
+  # On Rails 6.1, the VALID_OPTIONS array has a `freeze` property, so we cannot include the extra options
+  # I opted to override the `valid_options` method including the inherit options on it
+  module ActiveRecord::Associations::Builder
+    class Association
+      INHERIT_OPTIONS = %i[inherit inherit_allowed_list].freeze
+
+      def self.valid_options(options)
+        VALID_OPTIONS + INHERIT_OPTIONS + Association.extensions.flat_map(&:valid_options)
+      end
+    end
+  end
```

Comment:
Instead of monkey patching, I would prefer going with the module-prepend approach that (most of) the rest of this file uses. You could replace all of lines 3–29 with
```rb
module ActiveRecordInheritBuildAssocPrepend
  INHERIT_OPTIONS = %i[inherit inherit_allowed_list].freeze

  case ActiveRecord::VERSION::MAJOR
  when 4
    def valid_options
      super + INHERIT_OPTIONS
    end
  when 5, 6
    def valid_options(options)
      super + INHERIT_OPTIONS
    end
  end
end

ActiveRecord::Associations::Builder::Association.singleton_class.prepend(ActiveRecordInheritBuildAssocPrepend)
```

(I don’t particularly like the name `ActiveRecordInheritBuildAssocPrepend`, I just tried to be consistent with the existing naming scheme)

Reply:
Thank you so much for the nice suggestion. I tried something like that for the Rails 6 only, but I guess that the trick to make it work was on the `.singleton_class.prepend` part. Nice catch! 😁

Reply:
Patch instance methods with `Foo.prepend`; patch class methods with `Foo.singleton_class.prepend` 😉